jecs.luau

```diff
@@ -945,6 +975,13 @@ local function world_set(world: ecs_world_t, entity: i53, id: i53, data: unknown
 	end
 
 	if from then
+		if ECS_IS_PAIR(id) and bit32.band(idr.flags, ECS_ID_EXCLUSIVE) ~= 0 then
```

Comment:
it is probably cheap enough to just check the flags, you can benchmark and confirm. 

But if you want to check if it is a pair first, you should only look up the id_record *if* the id is a pair.